app/components/administrative-body-select.js

```diff
@@ -45,9 +56,24 @@ export default class AdministrativeBodySelectComponent extends Component {
       'sort': '-binding-start'
     });
 
-    this.administrativeBodyOptions = administrativeBodiesInTime.filter((administrativeBodyInTime) => {
-      let classificationUrl = administrativeBodyInTime.get('isTijdsspecialisatieVan.classificatie.uri');
-      return VALID_ADMINISTRATIVE_BODY_CLASSIFICATIONS.includes(classificationUrl);
-    });
+    this.administrativeBodyOptions = administrativeBodiesInTime.filter(
+      (administrativeBodyInTime) => {
+        const classificationUrl = administrativeBodyInTime.get(
+          'isTijdsspecialisatieVan.classificatie.uri'
+        );
+        const bodyIsValid =
+          VALID_ADMINISTRATIVE_BODY_CLASSIFICATIONS.includes(classificationUrl);
+        // shortcutting to avoid work
+        if (!bodyIsValid) {
+          return false;
+        }
+
+        const endDate = administrativeBodyInTime.bindingEinde;
+        if(this.args.endDateFilter) {
```

Comment:
This seems oddly specific and a bit of a weird component design tbh. Is there a reason why this can't be part of the component? This seems like a premature "keep things generic" optimization. I think this component is only used once, right?

Reply:
I agree, I think even if it's reused it's very likely the same filter will be used.

Reply:
fair enough, just thought it might be used somewhere else 
but even then the current design is odd, if we want a configurable filter it should receive the whole mandatee